src/charm.py

```diff
@@ -858,6 +891,19 @@ def _get_hostname_from_unit(self, member: str) -> str:
 
     def _on_leader_elected(self, event: LeaderElectedEvent) -> None:
         """Handle the leader-elected event."""
+        # consider configured system user passwords
+        system_user_passwords = {}
+        if admin_secret_id := self.config.system_users:
+            try:
+                system_user_passwords = self.get_secret_from_id(secret_id=admin_secret_id)
+            except (ModelError, SecretNotFoundError) as e:
+                # only display the error but don't return to make sure all users have passwords
+                logger.error(f"Error setting internal passwords: {e}")
+                self.unit.status = BlockedStatus("Password setting for system users failed.")
+                event.defer()
+
+        # this list is not consistent with `SYSTEM_USERS` -> todo: check if correct
+        # backup-user is missing here, patroni-user is missing in `SYSTEM_USERS`
```

Comment:
Patroni user is not a database user, but the REST API user for Patroni. We should most probably use a separate list like in PG VM, but this doesn't need to happen as part of this PR.